libunwind/src/UnwindRegistersRestore.S

```diff
@@ -703,6 +703,25 @@ Lnovec:
 
 #elif defined(__aarch64__)
 
+#if defined(__CHERI_PURE_CAPABILITY__)
+DEFINE_LIBUNWIND_FUNCTION(__rtld_unw_setcontext)
```

Comment:
Comment its prototype above like below?

Reply:
Also, does this really need to be in assembly? rtld's overriding implementation isn't.

Reply:
rtld's overriding implementation is in C because it calls into `unwind_stack()`, but the non-c18n implementation tail calls into the same assembly code as this (https://github.com/CTSRD-CHERI/cheribsd/blob/dev/libexec/rtld-elf/rtld_c18n.c#L856-L859). It's not great, but it was the simplest way to get the values in the right registers without trying to come up with some kind of ABI between libunwind and rtld. It's not great and should probably be reworked at some point in the future, perhaps with the CHERI-RISC-V implementation, but I probably don't have time to do that for the release.